### PR TITLE
[SPARK-36377][DOCS] Re-document "Options read in YARN client/cluster mode" section in spark-env.sh.template

### PR DESCRIPTION
**Pull request by yutoacts**
### What changes were proposed in this pull request?

Edit spark-env.sh.template for better documentation.

### Why are the changes needed?

Some options  (e.g. SPARK_CONF_DIR, SPARK_EXECUTOR_CORES, etc.) are read by any mode but are in the "Options read in YARN client/cluster mode" that might confuse users, so we should separate YARN only options from others.

### Does this PR introduce _any_ user-facing change?

Yes, docs changed.

### How was this patch tested?

Just a manual checking.
options

### CODE REVIEW

File: conf/spark-env.sh.template [hunk 1]
```diff
@@ -32,14 +32,16 @@
...
+# Options read in any mode
 # - SPARK_CONF_DIR, Alternate conf dir. (Default: ${SPARK_HOME}/conf)
-# - HADOOP_CONF_DIR, to point Spark towards Hadoop configuration files
-# - YARN_CONF_DIR, to point Spark towards YARN configuration files when you use YARN
 # - SPARK_EXECUTOR_CORES, Number of cores for the executors (Default: 1).
 # - SPARK_EXECUTOR_MEMORY, Memory per Executor (e.g. 1000M, 2G) (Default: 1G)
 # - SPARK_DRIVER_MEMORY, Memory for Driver (e.g. 1000M, 2G) (Default: 1G)
 
+# Options read in YARN client/cluster mode
+# - HADOOP_CONF_DIR, to point Spark towards Hadoop configuration files
```
Comment (HyukjinKwon) [inline conf/spark-env.sh.template:42]: Hm, I think actually this can be read in non Yarn modes too , e.g.) using HDFS in a different cluster mode

Comment (yutoacts) [reply conf/spark-env.sh.template:42]: Thanks for the review. I just fixed it and hope it makes sense.

**Comment by yutoacts** [comment]
@tgravescs Could you take a look? Thanks in advance.

File: conf/spark-env.sh.template [hunk 2]
```diff
@@ -32,14 +32,18 @@
...
-# Options read in YARN client/cluster mode
+# Options read in any mode
 # - SPARK_CONF_DIR, Alternate conf dir. (Default: ${SPARK_HOME}/conf)
-# - HADOOP_CONF_DIR, to point Spark towards Hadoop configuration files
-# - YARN_CONF_DIR, to point Spark towards YARN configuration files when you use YARN
 # - SPARK_EXECUTOR_CORES, Number of cores for the executors (Default: 1).
 # - SPARK_EXECUTOR_MEMORY, Memory per Executor (e.g. 1000M, 2G) (Default: 1G)
 # - SPARK_DRIVER_MEMORY, Memory for Driver (e.g. 1000M, 2G) (Default: 1G)
 
+# Options read in any cluster mode using HDFS (e.g. YARN)
```
Comment (tgravescs) [inline conf/spark-env.sh.template:41]: this is more then just cluster mode, it defines your default filesystem for instance, client mode will use it as well.

Comment (tgravescs) [reply conf/spark-env.sh.template:41]: also why the YARN bit, you could use HDFS in standalone mode for instance

Comment (yutoacts) [reply conf/spark-env.sh.template]: Thanks for the review. You're right, I meant cluster mode by any cluster manager but it must be confusing with YARN cluster mode. I fixed it, hope it makes more sense.

**Review by tgravescs — APPROVED** [review]
lgtm

**Comment by yutoacts** [comment]
@tgravescs Thanks.
@HyukjinKwon Does it look good to you?

**Review by HyukjinKwon — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**